### ISSUE TITLE
Links that open in new tabs/windows should be tagged for assistive technology

### ISSUE THREAD

**Issue by rebekaburnett** [facebook/docusaurus#10654]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Accessibility requirement:

Links that open in a new tab or a new window should either have an "aria-label" attribute or a screen-reader only element explaining to the screen-readers that this opens in a new tab.

This happens on every link that goes to another website. Here is an example: <a href="https://example.com/" target="_blank" rel="noopener noreferrer" class="footer__link-item" >Example page</a> (see attachment)

<img width="933" alt="1_links-open-tabs" src="https://github.com/user-attachments/assets/6ee772df-4fcb-490c-b9e8-0c8f988464e1">

### Reproducible demo

You can see this on https://docusaurus.io/

### Steps to reproduce

You can see the error by scanning the Docusaurs.io site with accessible: https://accessibe.com/accessscan?website=https://docusaurus.io/

### Expected behavior

Pass accessibility scans

### Actual behavior

Fails scans

### Your environment

- Public site URL: https://developer.vippsmobilepay.com/ also https://docusaurus.io/
- Docusaurus version used: 3.6

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by rebekaburnett** [comment]
It seems like these elements just need an aria-label like this:

```
<a href="https://example.com" target="_blank" aria-label="Example Link (opens in a new tab)">
  Example Link
</a>
```

**Comment by slorber** [comment]
Thanks for reporting a11y issues @rebekaburnett.

Now is not the right time for me, but I'll try to implement all the pending suggested improvements later.

**Comment by andrewnicols** [comment]
Rather than adding an aria-label, or perhaps in addition to, it would be great to make the behaviour configurable and have it _not_ open in a new window.

This has [previous been reported](https://docusaurus.io/feature-requests/p/external-links-opening-in-new-tabs-should-be-optional-a11y) but I can't see a GH issue for it.

**Comment by WestonThayer** [comment]
Noting that in terms of [WCAG](https://www.w3.org/TR/WCAG22/), there's no explicit requirement for links with `target="_blank"` to communicate that via `aria-label` or similar, **but** since there is a little visual arrow icon there currently:

<img width="94" height="26" alt="Privacy link with a little arrow icon after it." src="https://github.com/user-attachments/assets/f4496dab-f13d-4bf6-ab65-1235b3b3092b" />

[1.1.1 Non-Text Content](https://www.w3.org/TR/WCAG22/#non-text-content) kicks in, which it fails. Essentially, a sighted user can perceive the icon and understand that clicking this link will open in a new tab. Since the icon is marked `aria-hidden="true"` with no alternative text that feeds up into the link's accessible name, a blind user relying on a screen reader wouldn't be able to perceive equivalent information (aside: you could wonder why `target="_blank"` isn't a programmatic signal screen readers could surface for this, I'm sure there's an ARIA issue out there for it somewhere).

**Comment by slorber** [comment]
In https://github.com/facebook/docusaurus/pull/11331, we are going to add an `aria-label="(opens in new tab)"`, like this issue requested.

---

In https://github.com/facebook/docusaurus/pull/11257, we had a long discussion of the possibility of exposing an attribute to open external links in `_self` (eventually making this the default behavior). We couldn't reach a clear consensus, so I closed the PR. The aria-label should be enough for now, but we can continue discussing this behavior. 

If we ever take a decision to change the current link target behavior, I want this to be based on real data.

For non-disabled users, opening external links in `_self` can lead to a disappointing UX: https://x.com/sebastienlorber/status/1941065365125214281 . I suspect this would be less disappointing if we remove the external link icon, but that icon is something that users actually want, and many docs solutions implement, so removing it also degrades UX.

So, changing the behavior and/or removing the icon is not ideal and degrates the experience of sighted users.

We have at least one blind person that reported preferring opening links in new tabs: https://github.com/facebook/docusaurus/pull/11257#issuecomment-3090271591

If you want to change this behavior, please present us with the results of a survey of blind people that clearly show that the aria-label solution is not good enough, and that they would clearly prefer opening in `_self`. We can't degrade the experience of sighted users without even being sure that this improves the experience of blind users. 

I'd like to favor pragmatism and real-world data instead of theories and opinions.
